extension-style-app/src/androidTest/java/com/mapbox/maps/testapp/style/sources/generated/GeoJsonSourceTest.kt

```diff
@@ -75,24 +75,64 @@ class GeoJsonSourceTest : BaseStyleTest() {
   }
 
   @Test
-  @UiThreadTest
   fun urlTest() {
+    val latch = CountDownLatch(2)
+    val answerList = mutableListOf<String?>()
     val testSource = geoJsonSource(SOURCE_ID) {
       url(TEST_URI)
     }
-    setupSource(testSource)
-    assertEquals(TEST_URI, testSource.data)
+    val listener = OnSourceDataLoadedListener {
+      if (it.type == SourceDataType.METADATA && it.id == SOURCE_ID) {
+        answerList.add(testSource.data)
+        latch.countDown()
+      }
+    }
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapboxMap.apply {
+          addOnSourceDataLoadedListener(listener)
+          setupSource(testSource)
+        }
+      }
+    }
+    if (!latch.await(LATCH_MAX_TIME_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+    assertEquals(2, answerList.size)
+    assertEquals("", answerList[0])
+    assertEquals(TEST_URI, answerList[1])
+    mapboxMap.removeOnSourceDataLoadedListener(listener)
   }
 
   @Test
-  @UiThreadTest
   fun urlAfterBindTest() {
+    val latch = CountDownLatch(2)
+    val answerList = mutableListOf<String?>()
     val testSource = geoJsonSource(SOURCE_ID) {
       url(TEST_URI)
     }
-    setupSource(testSource)
-    testSource.url(TEST_URI)
-    assertEquals(TEST_URI, testSource.data)
+    val listener = OnSourceDataLoadedListener {
+      if (it.type == SourceDataType.METADATA && it.id == SOURCE_ID) {
+        answerList.add(testSource.data)
+        latch.countDown()
+      }
+    }
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapboxMap.apply {
+          addOnSourceDataLoadedListener(listener)
+          setupSource(testSource)
+          testSource.url(TEST_URI)
```

Comment:
1. Should we use another URI instead of the same to make sure that the one we're getting in L134 is the latest one?
2. Also, should there be some delay to this call to let the bindTo really set the initial TEST_URI?

Reply:
I guess we can set empty geoJson first and then apply `TEST_URI`, that will be better.